src/L.Control.Locate.js

```diff
@@ -303,6 +316,10 @@ You can find the project at: https://github.com/domoritz/leaflet-locatecontrol
                 this._map.on('locationerror', this._onLocationError, this);
                 this._map.on('dragstart', this._onDrag, this);
                 this._map.on('zoomstart', this._onZoom, this);
+                this._map.on('zoomend', this._onZoomEnd, this);
+                if (this.options.showCompass) {
+                    L.DomEvent.on(window, 'deviceorientation', this._onDeviceOrientation, this);
```

Comment:
Doesn't this need to be `deviceorientationabsolute`?

Reply:
Hmm, I don't think so. I'm mostly just porting code from that other leaflet plugin which is admittedly a little old, but deviceorientationabsolute doesn't appear to have a proper spec and doesn't seem to have much cross browser support.

https://developer.mozilla.org/en-US/docs/Web/API/Window/ondeviceorientationabsolute

deviceorientation has a proper spec, is 8 years old and has pretty good support. There are implementation differences but they don't seem to matter for our exact use case:

https://caniuse.com/#feat=deviceorientation

Reply:
I'm asking because I read https://developers.google.com/web/updates/2016/03/device-orientation-changes and it says that "Starting with Chrome 50, the degrees included in the DeviceOrientationEvent are by default no longer absolute with respect to the Earth’s coordinate frame." 

I have not yet tested your branch on my phone and I also don't know what the standard is.

Reply:
I have an android phone and can test it but I suggest you google first what others say about this. I’ll test it before I merge the PR.

Reply:
I've found a few bits of pieces:

https://gist.github.com/mikaelbr/0fed772d49fe655186a4e6ef4b270481#file-orientation-js-L51-L89

https://github.com/ai/compass.js/blob/master/lib/compass.js#L94-L108

But even that is after the chrome 50 release and doesn't seem to handle the absolute stuff. 

I did end up finding a spec which also has some js code for calculating turning the 3 vectors into a proper compass bearing. But I just don't trust any of this as I can't test it myself.

https://w3c.github.io/deviceorientation/#worked-example

Lastly there is a different api which is sorta the same related 

https://developer.mozilla.org/en-US/docs/Web/API/Coordinates/heading

The heading here is not quite what we want and is inferred from large scale moment, so you could be heading north in a car but the device is facing west so this isn't what we want. Most of the time this is null if you are not moving anyway. Even if we did want this we'd need to patch leaflet core as this isn't passed through the normalized locationfound event we get from there.

For now I have left the code as is and not tried to handle anything beyond what I can actually test. Sorry but I've run out of time for this so I can't do too much more. If you are not happy with the android side then I'd suggest just to not show the compass at all if we don't trust the data in certain cases and then build up support as it gets tested.

Reply:
I'll test it on my android tonight and merge if it works.